cli/src/cli_util.rs

```diff
@@ -522,6 +522,8 @@ impl ReadonlyUserRepo {
     }
 }
 
+/// Type-safe helper for conditionally advancing bookmarks
+///
 /// A bookmark that should be advanced to satisfy the "advance-bookmarks"
 /// feature. This is a helper for `WorkspaceCommandTransaction`. It provides a
```

Comment:
nit: insert blank line after "... feature." instead?

"helper for ..." sounds like an object that advances bookmark, but this is just a data object.